src/ripple/ledger/PaymentSandbox.h

```diff
@@ -0,0 +1,167 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_LEDGER_PAYMENTSANDBOX_H_INCLUDED
+#define RIPPLE_LEDGER_PAYMENTSANDBOX_H_INCLUDED
+
+#include <ripple/ledger/RawView.h>
+#include <ripple/ledger/Sandbox.h>
+#include <ripple/ledger/detail/ApplyViewBase.h>
+#include <ripple/protocol/AccountID.h>
+#include <map>
+#include <utility>
+
+namespace ripple {
+
+namespace detail {
+
+class DeferredCredits
```

Comment:
Its a detail class and not something anyone should be calling, except as required for the implementation of `PaymentView`. I'm okay with making it a nested declaration. Or better yet, inlining the behavior into `PaymentView` itself and eliminating the class. It evolved to where it is now, it wasn't designed to be part of the payment view initially. `PaymentView` explains its operation.